File: src/main/java/com/tibiawiki/process/RetrieveMounts.java
package com.tibiawiki.process;

import com.tibiawiki.domain.RetrieveAnyService;
import com.tibiawiki.domain.enums.InfoboxTemplate;
import com.tibiawiki.domain.factories.JsonFactory;
import com.tibiawiki.domain.mediawiki.ArticleRepository;
import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.stream.Collectors;

@Component
public class RetrieveMounts extends RetrieveAnyService {

    @Autowired
    public RetrieveMounts(ArticleRepository articleRepository, JsonFactory jsonFactory) {
        super(articleRepository, jsonFactory);
    }

    public List<String> getMountsList() {
        final List<String> mountsCategory = articleRepository.getPageNamesFromCategory(InfoboxTemplate.MOUNT.getCategoryName());
        final List<String> listsCategory = articleRepository.getPageNamesFromCategory(CATEGORY_LISTS);

        return mountsCategory.stream()
                .filter(page -> !listsCategory.contains(page))
                .collect(Collectors.toList());
    }

    public List<JSONObject> getMountsJSON() {
        return getArticlesFromInfoboxTemplateAsJSON(getMountsList());
    }

    public JSONObject getMountJSON(String pageName) {
        return super.getArticleAsJSON(pageName);
    }
}

File: src/main/kotlin/com/tibiawiki/process/RetrieveMounts.kt
package com.tibiawiki.process

import com.tibiawiki.domain.enums.InfoboxTemplate
import com.tibiawiki.domain.factories.ArticleFactory
import com.tibiawiki.domain.factories.JsonFactory
import com.tibiawiki.domain.repositories.ArticleRepository
import org.json.JSONObject
import org.springframework.stereotype.Component
import java.util.Optional
import java.util.stream.Stream

@Component
class RetrieveMounts(
    articleRepository: ArticleRepository,
    articleFactory: ArticleFactory,
    jsonFactory: JsonFactory
) : RetrieveAny(articleRepository, articleFactory, jsonFactory) {

    val mountsList: List<String>
        get() {
            val category = articleRepository.getPageNamesFromCategory(InfoboxTemplate.MOUNT.categoryName)
            val listsCategory = articleRepository.getPageNamesFromCategory(CATEGORY_LISTS)
            return category.filter { page -> page !in listsCategory }
        }

    val mountsJSON: Stream<JSONObject>
        get() = getArticlesFromInfoboxTemplateAsJSON(mountsList)

    fun getMountJSON(pageName: String): Optional<JSONObject> {
        return getArticleAsJSON(pageName)
    }
}
